torch/distributions/bijectors.py

```diff
@@ -0,0 +1,79 @@
+from torch.distributions.utils import broadcast_all
+
+
+class Bijector:
+    """
+    Abstract class `Bijector`. `Bijector` are bijective transformations with computable
+    inverse log det jacobians. They are meant for use in `TransformedDistribution`.
+    """
+    is_injective = True
+    add_inverse_to_cache = False
+
+    def __init__(self):
+        self._intermediates_cache = {}
+
+    def __call__(self, base_distribution):
+        """
+        Applies the bijector to the input distribution. Returns a TransformedDistribution.
+        """
+        from torch.distributions.transformed_distribution import TransformedDistribution
+        return TransformedDistribution(base_distribution, self)
+
+    def forward(self, value):
+        """
+        Invokes the bijection x=>y
+        """

```

Comment:
I think we should move caching behavior from `TransformedDistribution` to here. It seems more natural here.